<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>500</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>MQTT flow simulator</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="main_hlayout" stretch="1,0,1">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="left_vlayout">
        <item>
         <layout class="QHBoxLayout" name="broker_info_vlayout">
          <item>
           <widget class="QLabel" name="broker_lbl">
            <property name="text">
             <string>Broker</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="broker_hostname">
            <property name="toolTip">
             <string>Broker hostname</string>
            </property>
            <property name="text">
             <string>localhost</string>
            </property>
            <property name="placeholderText">
             <string>hostname</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="broker_port">
            <property name="toolTip">
             <string>Broker port (1-65535)</string>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>1883</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item alignment="Qt::AlignRight">
         <widget class="QPushButton" name="broker_connect_btn">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>Connect to broker</string>
          </property>
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="broker_line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="topics_vlayout">
          <property name="bottomMargin">
           <number>1</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="topics_hlayout" stretch="0,1">
            <item>
             <widget class="QLabel" name="topics_lbl">
              <property name="text">
               <string>Topics</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignRight">
             <widget class="QPushButton" name="add_topic_btn">
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="toolTip">
               <string>Add topic</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QScrollArea" name="topics_list_scroll_area">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="topics_list_widget">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>310</width>
               <height>324</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <layout class="QVBoxLayout" name="topics_list" stretch="0">
                <item>
                 <spacer name="topics_list_spacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="center_line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="logs_vlayout">
        <item>
         <widget class="QLabel" name="logs_lbl">
          <property name="text">
           <string>Logs</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="logs_list">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustIgnored</enum>
          </property>
          <property name="textElideMode">
           <enum>Qt::ElideNone</enum>
          </property>
          <property name="isWrapping" stdset="0">
           <bool>false</bool>
          </property>
          <property name="uniformItemSizes">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="sortingEnabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignRight">
         <widget class="QPushButton" name="clear_logs_btn">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>Clear logs</string>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
